### DISCUSSION TITLE
[New preprocessor] The "tile_colorfix+sharp" is added

### DISCUSSION THREAD

**Discussion by lllyasviel** [Mikubill/sd-webui-controlnet#1435] [General]
**1.1.196** added "tile_colorfix+sharp"
This method allows you to control the latent sharpness of the outputs of ControlNet tile.
This preprocessor can prevent the Tile model from the tendency to create the somewhat blurred "harmonious atmosphere" in your upscaled images.

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/5fbf68fa-eb8c-483c-b108-9925be1e6036)

# comparison 1

Input image (480x640)
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/d61ce438-a0de-4c61-8a6c-3dffafccad75)

Meta (Using T2I or I2I with 100% denoising strength, no extra upscaler/super-resolution, all results use same seed 12345):

1girl, best quality
Negative prompt: lowres, bad anatomy, bad hands, cropped, worst quality
Steps: 20, Sampler: Euler a, CFG scale: 7, Seed: 12345, Size: 960x1280, Model hash: 8712e20a5d, Model: Anything-V3.0, Clip skip: 2, Version: v1.2.1, ControlNet 0: "preprocessor: tile_XXXX, model: control_v11f1e_sd15_tile [a371b31b], weight: 1, starting/ending: (0, 1), resize mode: Crop and Resize, pixel perfect: True, control mode: Balanced, preprocessor params: (64, 8, 1)"

**tile_resample**:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/fd382840-8b6b-4358-848b-986cfdbd67aa)

**tile_colorfix**:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/c701d955-edf3-46b9-be84-8d1d40cbe56a)

**tile_colorfix+sharp (sharpness=0.5)**:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/739ac576-f4cb-44f4-b81e-dd733c8d21b6)

**tile_colorfix+sharp (default, sharpness=1.0)**:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/ec2356b6-d82f-477e-8ef7-275ee324bf70)

**img2img with 0.25 denoising strength without using any ControlNet**:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/c21b3e0e-8d45-4748-afb4-a84523c59949)

# comparison 2 (the difference for realistic image is less obvious)

Input image (480x640)
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/c9c4e9a2-f260-48f9-aa6c-436286e005ea)

Meta (Using T2I or I2I with 100% denoising strength, no extra upscaler/super-resolution, all results use same seed 12345):
old man, best quality
Negative prompt: lowres, bad anatomy, bad hands, cropped, worst quality
Steps: 20, Sampler: Euler a, CFG scale: 7, Seed: 12345, Size: 960x1280, Model hash: c0d1994c73, Model: realisticVisionV20_v20, Version: v1.2.1, ControlNet 0: "preprocessor: tile_XXXX, model: control_v11f1e_sd15_tile [a371b31b], weight: 1, starting/ending: (0, 1), resize mode: Crop and Resize, pixel perfect: True, control mode: Balanced, preprocessor params: (64, 8, 1)"

**tile_resample**:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/88c52bf3-295d-4887-8fd0-939e13728514)

**tile_colorfix**:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/47265ec7-eeb4-444c-a2ca-3e79e59a72d1)

**tile_colorfix+sharp (sharpness=0.5)**:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/90ed17c1-fe14-40c2-a465-d06d400801f8)

**tile_colorfix+sharp (default, sharpness=1.0)**:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/b3d001ab-ada3-4d82-bae1-7ace5b6c32f2)

**img2img with 0.25 denoising strength without using any ControlNet**:
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/70b4d9d8-4bb7-4cca-8f81-e17545f2688b)

**Note that all above results in this post are not cherry picked (they all use seed 12345). It is possible to get much better results using these preprocessors in other workflows (e.g., like using r-esrgan or other sr methods).**

# version
You need at least **1.1.196** to use it.

**Comment by lllyasviel** [comment]
Note that after 1.1.196, **tile_colorfix+sharp** becomes ControlNet's new best method in upscaling images.

**Comment by lllyasviel** [comment]
"tile_colorfix+sharp" requires less parameter tuning to apply to other models.
This is CounterfeitV30
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/cf1b1036-f745-4760-a94a-410c887693f0)
1girl, black_bow, flower, bow, long_hair, plant, solo, blue_eyes, outdoors, skirt, long_sleeves, white_shirt, two_side_up, breasts, looking_at_viewer, grass, shirt
Negative prompt: lowres, bad anatomy, bad hands, text, error, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality, normal quality, jpeg artifacts, signature, watermark, username, blurry
Steps: 20, Sampler: Euler a, CFG scale: 7, Seed: 525096802, Size: 960x1280, Model hash: cbfba64e66, Model: CounterfeitV30_v30, Clip skip: 2, ENSD: 31337, Version: v1.2.1, ControlNet 0: "preprocessor: tile_colorfix+sharp, model: control_v11f1e_sd15_tile [a371b31b], weight: 1, starting/ending: (0, 1), resize mode: Crop and Resize, pixel perfect: True, control mode: Balanced, preprocessor params: (64, 8, 1)"
(prompt is from Danbooru interrogator this time)

**Reply by lllyasviel** [reply]
From "Steps: 20, Sampler: Euler a, CFG scale: 7" we can see that it works well for common parameters without requiring much tuning

**Comment by Hosiokaa** [comment]
Could be a Me problem but I'm having weird color issues shown here.
![imgonline-com-ua-twotoone-hxMjp8kvjwWh](https://github.com/Mikubill/sd-webui-controlnet/assets/118495208/04e15458-c2c9-4985-a9cb-641dcf4ed674)

Right = Original
Left = ControlNet Tile colorfix + sharp (same happens on colorfix too)
Original image was 1280x1664

1girl, grey eyes, solo, dark atmosphere, outdoors, cloud, short hair, long sleeves, sky, black hair, cloudy sky, brown eyes, androgynous, standing, tree, cowboy shot, headband, closed mouth, grass
Negative prompt: (worst quality, low quality:1.2), lowres, bad anatomy, bad hands, nsfw, realistic
Steps: 30, Sampler: Euler a, CFG scale: 7, Seed: 88027789, Size: 1576x2048, Model hash: ca455ce14f, Model: Baka_Baka-DiffusionV4A(FP16)outattxn, Clip skip: 2, Version: v1.2.1, ControlNet 0: "preprocessor: tile_colorfix+sharp, model: control_v11f1e_sd15_tile [a371b31b], weight: 1, starting/ending: (0, 1), resize mode: Crop and Resize, pixel perfect: True, control mode: Balanced, preprocessor params: (64, 8, 1)"

**Comment by Rkkss** [comment]
Getting `ValueError: ControlNet failed to use VAE. Please try to add `--no-half-vae`, `--no-half` and remove `--precision full` in launch cmd.`
when run tile+colorfix, tile+colorfix sharp with multidiffusion.
I don't have those command in launch. Tested with anime kl-f8 and mse 84000. 

The regular tile resample prep runs without problem.

**Reply by lllyasviel** [reply]
update all extensions and completely restart terminal

**Reply by Rkkss** [reply]
> update all extensions and completely restart terminal


I have to set batch size to 1 for it to run otherwise I'll get OOM.
with regular resample preprocessor, I can do 4.

![abc](https://github.com/Mikubill/sd-webui-controlnet/assets/127614350/ef50d99d-85cf-4c49-b819-d7a8bfc4f7d1)
![a1](https://github.com/Mikubill/sd-webui-controlnet/assets/127614350/9c62785e-6164-4a0f-b5b3-a92c23d1abae)

**Reply by lllyasviel** [reply]
yes it needs more computation

**Comment by BNP1111** [comment]
preprocessor tile_colorfix+sharp   cant used with inpainting model

get this error:
RuntimeError: The size of tensor a (9) must match the size of tensor b (4) at non-singleton dimension 1

**Comment by DamienCz** [comment]
good afternoon,i just try to use this funcion,but i met a question about  the variation ,could u tell me what's useful? compare with sharpness,what's value i should put usually?

**Comment by VladAndronik** [comment]
Would be grateful if you could point out how to use it in diffusers pipelines or how to implement it there?
